sdk/network/Azure.ResourceManager.Network/tests/Tests/BastionHostContainerTests.cs

```diff
@@ -0,0 +1,130 @@
+            _publicIPAddressIdentifier = ipLro.Value.Id;
+            StopSessionRecording();
+        }
+
+        [SetUp]
+        public async Task TestSetUp()
+        {
+            var client = GetArmClient();
+            _resourceGroup = await client.GetResourceGroup(_resourceGroupIdentifier).GetAsync();
+            VirtualNetwork vnet = await _resourceGroup.GetVirtualNetworks().GetAsync(_subnetIdentifier.Parent.Name);
+            _subnet = await vnet.GetSubnets().GetAsync(_subnetIdentifier.Name);
+            _publicIPAddress = await _resourceGroup.GetPublicIPAddresses().GetAsync(_publicIPAddressIdentifier.Name);
+        }
+
+        private async Task<BastionHost> CreateBastionHost(string bastionName)
+        {
+            BastionHostData data = new BastionHostData();
+            data.Location = TestEnvironment.Location;
+            BastionHostIPConfiguration ipConfig = new BastionHostIPConfiguration();
+            ipConfig.Name = Recording.GenerateAssetName("bastionIPConfig-");
+            ipConfig.Subnet = new Models.SubResource();
+            ipConfig.Subnet.Id = _subnet.Id;
+            ipConfig.PublicIPAddress = new Models.SubResource();
+            ipConfig.PublicIPAddress.Id = _publicIPAddress.Id;
+            data.IpConfigurations.Add(ipConfig);
+            var bastionLro = await _resourceGroup.GetBastionHosts().CreateOrUpdateAsync(bastionName, data);
+            return bastionLro.Value;
+        }
+
+        [RecordedTest]
+        public async Task BastionHostBaseTest()
+        {
+            // create Bastion
+            string bastionName = Recording.GenerateAssetName("bastion-");
+            BastionHost bastionHost = await CreateBastionHost(bastionName);
+            Assert.IsNotNull(bastionHost.Data);
+            Assert.AreEqual(bastionName, bastionHost.Data.Name);
+            Assert.AreEqual(TestEnvironment.Location, bastionHost.Data.Location);
+            Assert.AreEqual(0, bastionHost.Data.Tags.Count);
+
+            // Put Bastion
+            var bastionContainer = _resourceGroup.GetBastionHosts();
+            string newBastionIpconfigName = Recording.GenerateAssetName("bastionIPConfig-");
+            var bastionData = new BastionHostData()
+            {
+                Location = TestEnvironment.Location,
+            };
+            bastionData.IpConfigurations.Add(new BastionHostIPConfiguration()
+            {
+                Name = newBastionIpconfigName,
+                Subnet = new Models.SubResource() { Id = _subnet.Id },
+                PublicIPAddress = new Models.SubResource() { Id = _publicIPAddress.Id },
+            });
+            bastionData.Tags.Add(new KeyValuePair<string, string>("key", "value"));
+            var putBastionOpereation = await bastionContainer.CreateOrUpdateAsync(bastionName, bastionData);
+            await putBastionOpereation.WaitForCompletionAsync();
+
+            // Get Bastion
```

Comment:
Prefer if each of these can be its own test.  That is why I created the OneTimeSetup to create the env necessary to run each test without each test needing to create it from scratch each time.

By having more granular tests it helps us understand what broke when a test fails quicker vs everything being asserted in a single test.

Reply:
I have pushed a new commit to resolve the comment, because of `Multiple BastionHosts cannot be created.` i have to add new member parameter`_bastionName` to confirm that there is only one BastionHost for the same ResourceGroup.
please review, thanks!